components/feature/pwa/src/main/java/mozilla/components/feature/pwa/db/ManifestDao.kt

```diff
@@ -17,8 +17,8 @@ import androidx.room.Update
 @Dao
 internal interface ManifestDao {
     @WorkerThread
-    @Query("SELECT * from manifests WHERE start_url = :startUrl")
-    fun getManifest(startUrl: String): ManifestEntity?
+    @Query("SELECT * from manifests WHERE :url LIKE scope||'%' ORDER BY LENGTH(scope) DESC LIMIT 1")
+    fun getManifest(url: String): ManifestEntity?
```

Comment:
Let's make a separate query for getting the manifest using an arbitrary URL. That way we can continue to use the primary key when its known, such as in `updateManifest`. We should also make `scope` an index so its faster to query.

Reply:
@NotWoods scope should already be an index https://github.com/mozilla-mobile/android-components/pull/5724/commits/e673751375fc375269a9d8d43b19715753b53b7e#diff-a37ee58885a902172361ba486bd51783R23-R24

Reply:
Oops, missed that. Sorry!